modules/claims_api/lib/custom_error.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module ClaimsApi
+  class CustomError
+    def initialize(error, claim, method)
+      @error = error
+      @claim = claim
+      @method = method
+    end
+
+    def build_error # rubocop:disable Metrics/MethodLength
+      get_status
+      get_code
+      if @error == Faraday::ConnectionFailed || @error == Faraday::ParsingError ||
+         @error == Faraday::NilStatusError || @error == Faraday::TimeoutError ||
+         @error == ::Common::Exceptions::BackendServiceException ||
```

Comment:
Not sure this check works as expected.  Using the `==` did not match for the error I raised, however it appears using `is_a?(` did match

<img width="656" alt="Screenshot 2024-03-07 at 10 00 51 AM" src="https://github.com/department-of-veterans-affairs/vets-api/assets/129893414/4483e9e7-9dd5-4d1d-a837-edb958ac70d9">

Reply:
Which would be similar to what the old behavior did by formatting it as the 'source'

![Screenshot 2024-03-07 at 10 17 52 AM](https://github.com/department-of-veterans-affairs/vets-api/assets/129893414/f90b5440-d3b1-430f-9ea1-a5b147c281e2)